akd/configs/data_search_config.py

```diff
@@ -0,0 +1,266 @@
+"""
+Centralized configuration for data search components.
+"""
+
+import os
+from enum import Enum
+from typing import Optional
+
+from pydantic import BaseModel, Field, HttpUrl
+
+
+class Environment(str, Enum):
+    """Supported environments."""
+
+    DEVELOPMENT = "development"
+    STAGING = "staging"
+    PRODUCTION = "production"
+
+
+class WebSocketConfig(BaseModel):
+    """WebSocket connection configuration."""
+
+    host: str = Field(default="localhost", description="WebSocket host")
+    port: int = Field(default=8003, description="WebSocket port")
+    max_reconnect_attempts: int = Field(
+        default=5,
+        description="Maximum reconnection attempts",
+    )
+    base_reconnect_delay: int = Field(
+        default=1000,
+        description="Base reconnection delay in milliseconds",
+    )
+    connection_timeout: float = Field(
+        default=10.0,
+        description="Connection timeout in seconds",
+    )
+
+    @property
+    def url(self) -> str:
+        """Get the WebSocket URL."""
+        return f"ws://{self.host}:{self.port}"
+
+    @property
+    def http_base_url(self) -> str:
+        """Get the HTTP base URL."""
+        return f"http://{self.host}:{self.port}"
+
+
+class MCPConfig(BaseModel):
+    """MCP server configuration."""
+
+    endpoint: HttpUrl = Field(
+        default="http://localhost:8080/mcp/cmr/mcp/",
```

Comment:
can we have this set from os.getenv thing and default to localhost. That way we can control the config via env variable as well. See `searxng` etc